auxx/src/BlockGen.hs

```diff
@@ -0,0 +1,43 @@
+-- | Block generation.
+
+module BlockGen
+       ( generateBlocks
+       ) where
+
+import           Universum
+
+import           Control.Monad.Random.Strict (evalRandT)
+import           Data.Default                (def)
+import           System.Random               (mkStdGen, randomIO)
+
+import           Pos.AllSecrets              (mkAllSecretsSimple)
+import           Pos.Core                    (gdBootStakeholders, genesisData)
+import           Pos.Crypto                  (encToSecret)
+import           Pos.Generator.Block         (BlockGenParams (..), genBlocks)
+import           Pos.Launcher                (HasConfigurations)
+import           Pos.Wallet                  (getAllUserSecrets, getSecretKeys)
+
+import           Command.Types               (GenBlocksParams (..))
+import           Mode                        (AuxxMode)
+
+
+generateBlocks :: HasConfigurations => GenBlocksParams -> AuxxMode ()
+generateBlocks GenBlocksParams{..} = do
+    seed <- liftIO $ maybe randomIO pure bgoSeed
+    putText $ "Generating with seed " <> show seed
+
+    allSecrets <- mkAllSecretsSimple . map encToSecret . getAllUserSecrets <$> getSecretKeys
+
+    let bgenParams =
+            BlockGenParams
+                { _bgpSecrets         = allSecrets
+                , _bgpGenStakeholders = gdBootStakeholders genesisData
+                , _bgpBlockCount      = fromIntegral bgoBlockN
+                , _bgpTxGenParams     = def
```

Comment:
Why is it `def`? It should be configurable.

Reply:
1. It shouldn't be, task doesn't say anything about it
2. It's hard to make it configurable because parsing optionals is hard if you use parsec not optparse.

Reply:
The task says that you should move block-gen to auxx, _move_ means that you shouldn't throw-away existing features, they should be preserved.
The most important use case is to generate empty blocks, with `def` it's not achievable.